examples/example_app/example_app/views.py

```diff
@@ -38,6 +39,7 @@ class DeviceDetailAppTabOneView(views.ObjectView):
 
     queryset = Device.objects.all()
     template_name = "example_app/tab_device_detail_1.html"
+    object_detail_content = ui.ObjectDetailContent(extra_buttons=DeviceView.object_detail_content.extra_buttons)
```

Comment:
Maybe just buttons are a different type than `object_detail_content` - there are tied more with detail page as a whole rather than individual "tab" - custom or not. I can think of two possible solutions:
1. Create separate class field specifically for `page_buttons` and then all of the devices tab's should inherit that.  To still allow users to define custom buttons per tab we can add some logic into main view to combine `page_buttons` with those `extra_buttons` defined in give `object_detail_content`.
2. Create something like `ExtraButtonsMixin` and then `DevicePageButtonsMixin` and add it to all of the device related tabs. In this mixin you can add a logic, again, to merge `page_buttons` with `extra_buttons`.

Reply:
In theory `object_detail_content` defines the entire page rather than just the main tab, we just provide some shortcuts to make it easier to directly manipulate the main tab since that's the 80% use case. That's probably not clearly documented as a design goal though.

A couple other possibilities that occur to me:

- Require/recommend custom tabs for a detail view to inherit from the detail view class itself (e.g., `dcim.views.DeviceView`) rather than from `generic.ObjectView`. This wouldn't work for UIViewSet views though so we'd need a different pattern for those. Plus this makes app tabs more tightly coupled to the base view implementation such that converting a base view from an ObjectView to a UIViewSet would become a breaking change. So maybe not.
- Add another template filter used in `object_retrieve.html` that takes the `obj` from the context, does a `get_view_for_model(obj)` to get the base detail view, and injects the `extra_buttons` from that view automatically? I think this has some potential, I'll try and explore it today.